src/notifications.ts

```diff
@@ -411,31 +411,42 @@ export class Notifications implements StreamingClientExtension {
     if (response && response.data && 'entities' in response.data && Array.isArray(response.data.entities)) {
       topicResponseEntities = response.data.entities;
     }
-    const topicResponsesById: { [topic: string]: ChannelTopicResponseEntity } = {};
+    const result: BulkSubscribeResult = {};
     for (const topicEntity of topicResponseEntities) {
-      topicResponsesById[topicEntity.id] = topicEntity;
+      const { id, state, rejectionReason } = topicEntity;
+      const topicResult = result[id] = { topic: id, state, rejectionReason };
+      // If response entity is a combined topic ID like "a.b?c&d" include individualized topic IDs
+      // as keys in the map. This could either point to the same result as the combined topic ID
+      // or to a specific result for that individual topic if backend provides a specific result.
+      // Example: caller asked to subscribe "a.b?c&d" but user lacks permission for topic "a.b.d"
+      // In this case, API response will include "a.b?c&d" as success along with "a.b.d" as failure.
+      if (id.includes('?')) {
+        for (const individualTopic of splitIntoIndividualTopics(id)) {
+          const hasIndividualTopicResult = (individualTopic in result);
+          // Only use the combined topic result for this individual topic ID if there isn't already
+          // a result for the individual topic itself. Exact topic result takes precedence.
+          if (!hasIndividualTopicResult) {
+            result[individualTopic] = topicResult;
+          }
+        }
+      }
     }
-    const result: BulkSubscribeResult = {};
 
     if (options.replace) {
       this.bulkSubscriptions = {};
     }
 
     topics.forEach(topic => {
       this.bulkSubscriptions[topic] = true;
-
-      if (this.enablePartialBulkResubscribe) {
-        if (topic in topicResponsesById) {
-          const { state, rejectionReason } = topicResponsesById[topic];
-          result[topic] = { topic, state, rejectionReason };
-        } else {
-          result[topic] = { topic, state: 'Unknown' };
-        }
-      } else {
-        result[topic] = { topic, state: 'Permitted' };
-      }
```

Comment:
Part of why this was wrong is it was only looping through the bulk topics list `topics` rather than through all the requested topics which is what `toSubscribe` is. The other piece was to use the API response as more of the "source of truth" for the bulk result and then only add in stuff from `toSubscribe` below if it wasn't in the bulk response.